libcaf_core/caf/actor_system.hpp

```diff
@@ -491,6 +491,14 @@ class actor_system {
   /// Blocks the caller until all detached threads are done.
   void await_detached_threads();
 
+  /// Calls all thread started hooks
+  /// @warning must be called by thread which is about to start
+  void call_thread_started_cb();
```

Comment:
The past or continuous tense already implies a callback nature and I feel like the `call_` prefix is redundant. I'd simply rename this to `thread_started()` and the function below to `thread_terminates`.